Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -6822,6 +6822,14 @@ void WebPageProxy::didStartProvisionalLoadForFrameShared(Ref<WebProcessProxy>&&
     frame->setUnreachableURL(unreachableURL);
     frame->didStartProvisionalLoad(WTFMove(url));
 
+    if (m_controlledByAutomation) {
```

Comment:
You added the call to `navigationStartedForFrame()` but it's missing the guard `#if ENABLE(WEBDRIVER_BIDI)`, which is why tvOS build is failing.